lasagne/tests/layers/test_noise.py

```diff
@@ -155,3 +155,103 @@ def test_specified_rng(self, input_layer):
+
+    def test_get_output_for_deterministic(self, layer):
+        input = theano.shared(numpy.ones((100, 100)))
+        result = layer.get_output_for(input, deterministic=True)
+        result_eval = result.eval()
+        assert (result_eval == input.get_value()).all()
+
+    def test_get_output_for_no_rescale(self, layer_no_rescale):
+        input = theano.shared(numpy.ones((100, 100)))
+        result = layer_no_rescale.get_output_for(input)
+        result_eval = result.eval()
+        assert 0.8 < result_eval.mean() < 1.0
+
+    def test_get_output_for_no_rescale_dtype(self, layer_no_rescale):
+        input = theano.shared(numpy.ones((100, 100), dtype=numpy.int32))
+        result = layer_no_rescale.get_output_for(input)
+        assert result.dtype == input.dtype
+
+    def test_get_output_for_p_02(self, layer_p_02):
+        input = theano.shared(numpy.ones((100, 100)))
+        result = layer_p_02.get_output_for(input)
+        result_eval = result.eval()
+        assert 0.8 < result_eval.mean() < 1.0
+
+    def test_get_output_for_p_float32(self, input_layer):
+        from lasagne.layers.noise import AlphaDropoutLayer
+        layer = AlphaDropoutLayer(input_layer, p=numpy.float32(0.5))
+        input = theano.shared(numpy.ones((100, 100), dtype=numpy.float32))
+        assert layer.get_output_for(input).dtype == input.dtype
+
+    @pytest.mark.parametrize("shared_axes", [(), (0,), (2, 3), (-1, -2)])
+    def test_get_output_for_shared_axes(self, shared_axes):
+        from lasagne.layers.noise import AlphaDropoutLayer
+        layer = AlphaDropoutLayer((2, 4, 7, 9), shared_axes=shared_axes)
+        input = theano.shared(numpy.ones((2, 4, 7, 9)))
+        result = layer.get_output_for(input)
+        result_eval = result.eval()
+        # check if the dropout mask is the same across the specified axes:
+        # compute the mean across these axes and compare against the full
+        # output, broadcasting across the shared axes, to see if it matches
+        assert np.allclose(result_eval.mean(axis=shared_axes, keepdims=True),
+                           result_eval)
+
+    def test_specified_rng(self, input_layer):
+        from lasagne.layers.noise import AlphaDropoutLayer
+        input = theano.shared(numpy.ones((100, 100)))
+        seed = 123456789
+        rng = get_rng()
+
+        set_rng(RandomState(seed))
+        result = AlphaDropoutLayer(input_layer).get_output_for(input)
+        result_eval1 = result.eval()
+
+        set_rng(RandomState(seed))
+        result = AlphaDropoutLayer(input_layer).get_output_for(input)
+        result_eval2 = result.eval()
+
+        set_rng(rng)  # reset to original RNG for other tests
+        assert numpy.allclose(result_eval1, result_eval2)
+
+    def test_alpha(self, input_layer):
+        input = theano.shared(numpy.ones((100, 100)))
+        from lasagne.layers.noise import AlphaDropoutLayer
+        layer = AlphaDropoutLayer(input_layer, alpha=-1.7580993408473766)
```

Comment:
But that's just the default value for `alpha`. It should test another value, otherwise you never know whether that value was actually respected or it took the default anyway :)